deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,148 @@
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+from deepchecks.vision.vision_data import TaskType
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric, default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+    """
+
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+        if 'metric_name' in kwargs:
+            self.metric_name = kwargs['metric_name']
+
+        if 'reduce_name' in kwargs:
+            self.reduce_name = kwargs['reduce_name']
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'accuracy'
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'object_detection_average_precision'
+                if self.reduce is None:
+                    self.reduce = torch.nanmean
+            else:
+                raise DeepchecksValueError('For task types other then classification or object detection, '
+                                           'pass a metric explicitly')
+        self.metric.reset()
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        self.metric.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        metric_result = self.metric.compute()
+        if self.reduce is not None:
+            if isinstance(metric_result, float):
+                warnings.warn(SyntaxWarning('Metric result is already scalar, skipping reduce operation'))
```

Comment:
is there something the user can do to prevent this warning? something like 
"Metric result is already scalar, skipping reduce operation. Pass reduce=None to prevent this"

Reply:
maybe also check if it's a numpy float? (or maybe even a number/numpy number)

Reply:
Why specifically numpy?

Reply:
seems possible for a meric to return numpy no?

Reply:
`isinstance(metric_result, numbers.Number)` will catch both

Reply:
number.Number doesn't have a default conversion to float, we're expecting an ignite Metric so i don't see a case that will return numpy.

Reply:
ignite metrics can technically return anything

Reply:
changed to number.Real